crates/solver/src/settlement_submission.rs

```diff
@@ -98,6 +99,7 @@ impl SolutionSubmitter {
                             gas_estimate,
                             deadline: Some(Instant::now() + self.max_confirm_time),
                             retry_interval: self.retry_interval,
+                            try_to_recover_gas_price,
```

Comment:
Another alternative instead of specifying whether or not we should try to recover gas prices would be to always try, and silently fail if its not there.

Alternatively, it could be possible for a future private network API to have a request for recovering a users transaction requiring some PK signature (otherwise, what is the point to private transactions 🤣). So the other options would be to add a new method `TransactionSubmitting::recover_pending_transaction()` where for now the `CustomNodes` implementation uses the mem-pool, but `Flashbots` and `Eden` just return `None`.

Reply:
Refactored according to the above explanation.